### PR TITLE
Spawner-ZMQ

### PR DESCRIPTION
**Pull request by erezz-mov-ai**
[BP-255](https://movai.atlassian.net/browse/BP-255) - integrating the spawner

[BP-255]: https://movai.atlassian.net/browse/BP-255?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: flow_initiator/__init__.py [hunk 1]
```diff
@@ -38,5 +39,5 @@ def main():
         default=1,
     )
     args = parser.parse_args()
-    app = Core(args)
-    return app.run()
+    app = SpawnerManager(args)
+    return app.start()
```
Comment (engdorm) [inline flow_initiator/__init__.py:43]: shouldn't it be starup?

File: flow_initiator/spawner/async_manager.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            asyncio.create_task(self.core.spawner.process_command(command_dict))
+            response_msg = "Got request & successfully proccessed".encode("utf8")
+        except json.JSONDecodeError as e:
+            self._logger.error(f"can't parse command: {buffer}")
+            self._logger.error(e)
+            response_msg = "can't parse command: {buffer}".encode("utf8")
+        finally:
+            await self._socket.send_multipart(response_msg)
+
+    async def startup(self):
```
Comment (engdorm) [inline flow_initiator/spawner/async_manager.py:73]: it's startup 
and it's async

File: flow_initiator/spawner/async_movaicore.py [hunk 3]
```diff
@@ -173,39 +206,121 @@ async def callback(self, msg: tuple, connection: aioredis.Redis) -> None:
...
         ]
         list(map(lambda task: task.cancel(), tasks))
         await asyncio.gather(*tasks, return_exceptions=True)
 
         # clean enabled locks pool to finish any hanging threads
         Lock.enabled_locks = []
 
-    def run(self):
-
-        self.loop.run_until_complete(self.spin())
```
Comment (engdorm) [inline flow_initiator/spawner/async_movaicore.py:195]: what calls the spin?

Comment (erezz-mov-ai) [reply flow_initiator/spawner/async_movaicore.py:195]: ZMQServer initialize the loop

**Review by engdorm — APPROVED** [review]

**Pull request merged**